app/javascript/components/login/components/idp-selection/utils.js

```diff
@@ -1,73 +1,54 @@
+        clientId: idp.client_id,
+        authority: idp.authorization_url,
+        knownAuthorities: ["unicefpartners.b2clogin.com"],
+        validateAuthority: false,
+        redirectUri: idp.redirect_uri
+      },
+      cache: {
+        cacheLocation: "sessionStorage",
+        storeAuthStateInCookie: false
+      }
+    };
+  }
 
-export const setMsalConfig = idp => {
   return {
     auth: {
-      clientId: idp.get("client_id"),
-      authority: idp.get("authorization_url"),
+      protocolMode: ProtocolMode.OIDC,
+      clientId: idp.client_id,
+      authority: idp.authorization_url,
+      autoRefreshToken: true,
+      authorityMetadata: JSON.stringify({
+        authorization_endpoint: `${idp.authorization_url}/protocol/openid-connect/auth`,
+        token_endpoint: `${idp.authorization_url}/protocol/openid-connect/token`,
+        end_session_endpoint: `${idp.authorization_url}/protocol/openid-connect/logout`,
+        issuer: idp.authorization_url
+      }),
+      knownAuthorities: ["unicefpartners.b2clogin.com", idp.authorization_url],
       validateAuthority: false,
-      redirectUri: `${PROTOCOL}//${DOMAIN}/login/${idp.get("provider_type")}`
+      redirectUri: idp.redirect_uri
     },
     cache: {
-      cacheLocation: "localStorage",
-      storeAuthStateInCookie: true
+      cacheLocation: "sessionStorage",
+      storeAuthStateInCookie: false
     }
   };
 };
 
-export const setMsalApp = (msalConfig, forceStandardOidc) => {
-  if (forceStandardOidc) {
-    // This affects a couple of behaviours:
-    // The first is the form of the OIDC well-known configuration URI, to which msal appends a non-standard "v2.0"
-    // between the issuer and the well-known suffix (when this function returns false)
-    // See https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig
-    // The second is how msal creates its "client info" structure. If this is false, it expects a non-standard
-    // client_info parameter to be returned on the authorization response (standard params:
-    // https://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthResponse)
-    // When this returns true, it falls back on creating this structure from the id token, which succeeds
-    Authority.isAdfs = () => true;
-  }
-
-  const app = new UserAgentApplication(msalConfig);
-
-  if (forceStandardOidc) {
-    // A second situation where it expects a parameter that isn't standard, this time "scope"
-    // See https://openid.net/specs/openid-connect-core-1_0.html#ImplicitAuthResponse for standard params
-    // Luckily the scopes exist in a claim on the authorization token, which is passed to this function. We can
-    // intercept the method call, parse the auth token, and place the extracted scope in the parameters, where msal
-    // wants it
-    const oldSaveAccessToken = app.saveAccessToken.bind(app);
+export const setMsalApp = (msalConfig, historyObj) => {
+  const app = new PublicClientApplication(msalConfig);
```

Comment:
vad ca inainte era UserAgentApplication acum e PublicClientApplication, sigur nu trebuie pastrat cum era inainte pt ei?

Reply:
e de la ei schimbarea in PublicClientApplication, a fost din migrarea lor v1 -> v2